x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -0,0 +1,437 @@
+          const nodeId = `${nextLinkId}-${JOB_MAP_NODE_TYPES.INDEX_PATTERN}`;
+          result.elements.unshift({
+            data: { id: nodeId, label: nextLinkId, type: JOB_MAP_NODE_TYPES.INDEX_PATTERN },
+          });
+          result.details[nodeId] = link.indexData;
+
+          if (link.meta?.created_by === 'data-frame-analytics') {
+            nextLinkId = link.meta.analytics;
+            nextType = JOB_MAP_NODE_TYPES.ANALYTICS;
+          }
+
+          if (link.meta?.created_by === JOB_MAP_NODE_TYPES.TRANSFORM) {
+            nextLinkId = link.meta._transform?.transform;
+            nextType = JOB_MAP_NODE_TYPES.TRANSFORM;
+          }
+
+          // Check meta data
+          if (link.meta === undefined || link.meta?.created_by === INDEX_META_DATA_CREATED_BY) {
+            rootIndexPattern = nextLinkId;
+            complete = true;
+            break;
+          }
+        } else if (isJobDataLinkReturnType(link) && link.isJob === true) {
+          data = link.jobData;
+          const nodeId = `${data.id}-${JOB_MAP_NODE_TYPES.ANALYTICS}`;
+
+          result.elements.unshift({
+            data: {
+              id: nodeId,
+              label: data.id,
+              type: JOB_MAP_NODE_TYPES.ANALYTICS,
+              analysisType: getAnalysisType(data?.analysis),
+            },
+          });
+          result.details[nodeId] = data;
+          nextLinkId = data?.source?.index[0];
+          nextType = JOB_MAP_NODE_TYPES.INDEX_PATTERN;
+
+          // Get inference model for analytics job and create model node
+          ({ modelElement, modelDetails, edgeElement } = this.getAnalyticsModelElements(data.id));
+          if (isAnalyticsMapNodeElement(modelElement)) {
+            modelElements.push(modelElement);
+            result.details[modelElement.data.id] = modelDetails;
+          }
+          if (isAnalyticsMapEdgeElement(edgeElement)) {
+            modelElements.push(edgeElement);
+          }
+        } else if (isTransformLinkReturnType(link) && link.isTransform === true) {
+          data = link.transformData;
+
+          const nodeId = `${data.id}-${JOB_MAP_NODE_TYPES.TRANSFORM}`;
+          rootTransform = data.dest.index;
+
+          result.elements.unshift({
+            data: { id: nodeId, label: data.id, type: JOB_MAP_NODE_TYPES.TRANSFORM },
+          });
+          result.details[nodeId] = data;
+          nextLinkId = data?.source?.index[0];
+          nextType = JOB_MAP_NODE_TYPES.INDEX_PATTERN;
+        }
+      } // end while
+
+      // create edge elements
+      const elemLength = result.elements.length - 1;
+      for (let i = 0; i < elemLength; i++) {
+        const currentElem = result.elements[i];
+        const nextElem = result.elements[i + 1];
+        if (currentElem !== undefined && nextElem !== undefined) {
+          result.elements.push({
+            data: {
+              id: `${currentElem.data.id}~${nextElem.data.id}`,
```

Comment:
why do you use `~` as a separator here?

Reply:
Just a way to denote that the edge connects one node id to another node id. It doesn't have any particular significance.